player_darwin.go

```diff
@@ -0,0 +1,97 @@
+package oto
+
+// #cgo CFLAGS: -I/System/Library/Frameworks
+// #cgo LDFLAGS: -framework CoreAudio -framework AudioToolbox -framework Foundation
+// #include "player_darwin.h"
+import "C"
+
+// global buffer since it must be accessed from the go_input_callback
+var Buf []byte
```

Comment:
This isn't meant to be final. I wanted to have a conversation about how to deal with this issue. I need to have access to a buffer from a callback function that is exported to C, and we need to be able to write to that buffer in Go, and pass the buffer to C. I think it may be possible to handle this problem by writing a function that accepts a *player as the first parameter, which I can probably handle in C with a void*, but I haven't tried it yet. I'm open to ideas.